tests/lib/hardware/libvirt.py

```diff
@@ -72,7 +74,16 @@ def boot(self):
         logger.info(f"node {self.name}: booting with "
                     f"image {self._snap_img_path}")
         logger.debug(f"node {self.name}: libvirt xml: {xml}")
-        self._dom = self._conn.defineXML(xml)
+        try:
+            libvirtDefineNodeLock.acquire()
+            self._dom = self._conn.defineXML(xml)
+        except libvirt.libvirtError as e:
+            logger.error(f"unable to define node '{self.name}' using xml: {xml}")
+            raise e
+        finally:
+            logger.debug(f"Releasing node define lock")
```

Comment:
if you put that debug here, please also put it when getting the lock

Reply:
I've initially put the debug before the lock, but later removed since it was overkill, I would rather delete this one.